web/src/test/java/org/cbioportal/web/SampleListControllerTest.java

```diff
@@ -0,0 +1,256 @@
+            .andExpect(MockMvcResultMatchers.status().isNotFound())
+            .andExpect(MockMvcResultMatchers.jsonPath("$.message")
+                .value("Sample list not found: test_sample_list_id"));
+    }
+
+    @Test
+    public void getSampleList() throws Exception {
+
+        SampleList sampleList = new SampleList();
+        sampleList.setListId(TEST_LIST_ID_1);
+        sampleList.setStableId(TEST_STABLE_ID_1);
+        sampleList.setCancerStudyId(TEST_CANCER_STUDY_ID_1);
+        sampleList.setCancerStudyIdentifier(TEST_STUDY_IDENTIFIER_1);
+        sampleList.setCategory(TEST_CATEGORY_1);
+        sampleList.setName(TEST_NAME_1);
+        sampleList.setDescription(TEST_DESCRIPTION_1);
+        CancerStudy cancerStudy = new CancerStudy();
+        cancerStudy.setCancerStudyId(TEST_CANCER_STUDY_ID_1);
+        cancerStudy.setCancerStudyIdentifier(TEST_CANCER_STUDY_IDENTIFIER_1);
+        cancerStudy.setTypeOfCancerId(TEST_TYPE_OF_CANCER_ID_1);
+        cancerStudy.setName(TEST_STUDY_NAME_1);
+        cancerStudy.setShortName(TEST_SHORT_NAME_1);
+        cancerStudy.setDescription(TEST_STUDY_DESCRIPTION_1);
+        sampleList.setCancerStudy(cancerStudy);
+
+        Mockito.when(sampleListService.getSampleList(Mockito.anyString())).thenReturn(sampleList);
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/sample-lists/test_sample_list_id")
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(MockMvcResultMatchers.status().isOk())
+            .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.listId").doesNotExist())
+            .andExpect(MockMvcResultMatchers.jsonPath("$.sampleListId").value(TEST_STABLE_ID_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.cancerStudyId").value(TEST_STUDY_IDENTIFIER_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.category").value(TEST_CATEGORY_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.name").value(TEST_NAME_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.description").value(TEST_DESCRIPTION_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.cancerStudy.cancerStudyId").doesNotExist())
+            .andExpect(MockMvcResultMatchers.jsonPath("$.cancerStudy.cancerStudyIdentifier")
+                .value(TEST_CANCER_STUDY_IDENTIFIER_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.cancerStudy.description").value(TEST_STUDY_DESCRIPTION_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.cancerStudy.name").value(TEST_STUDY_NAME_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.cancerStudy.shortName").value(TEST_SHORT_NAME_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$.cancerStudy.typeOfCancerId")
+                .value(TEST_TYPE_OF_CANCER_ID_1));
+    }
+
+    @Test
+    public void getAllSampleListsInStudyDefaultProjection() throws Exception {
```

Comment:
Add web layer test for getAllSampleListsDetailedProjection() and getAllSampleListsInStudyDetailedProjection()